R/substitution.R

```diff
@@ -0,0 +1,609 @@
+#' Substitute missing values in the table body
+#'
+#' @description
+#' Wherever there is missing data (i.e., `NA` values) customizable content may
+#' present better than the standard `NA` text that would otherwise appear. The
+#' `sub_missing()` function allows for this replacement through its
+#' `missing_text` argument (where an em dash serves as the default).
+#'
+#' @details
+#' Targeting of values is done through `columns` and additionally by `rows` (if
+#' nothing is provided for `rows` then entire columns are selected). Conditional
+#' formatting is possible by providing a conditional expression to the `rows`
+#' argument. See the Arguments section for more information on this.
+#'
+#' @inheritParams fmt_number
+#' @param missing_text The text to be used in place of `NA` values in the
+#'   rendered table.
+#'
+#' @return An object of class `gt_tbl`.
+#'
+#' @section Examples:
+#'
+#' Use [`exibble`] to create a **gt** table. The `NA` values in different
+#' columns will be given replacement text with two calls of `sub_missing()`.
+#'
+#' ```r
+#' exibble %>%
+#'   dplyr::select(-row, -group) %>%
+#'   gt() %>%
+#'   fmt_missing(
+#'     columns = 1:2,
+#'     missing_text = "missing"
+#'   ) %>%
+#'   fmt_missing(
+#'     columns = 4:7,
+#'     missing_text = "nothing"
+#'   )
+#' ```
+#'
+#' \if{html}{\out{
+#' `r man_get_image_tag(file = "man_sub_missing_1.png")`
+#' }}
+#'
+#' @family Format Data
+#' @section Function ID:
+#' 3-16
+#'
+#' @import rlang
+#' @export
+sub_missing <- function(
+    data,
+    columns = everything(),
+    rows = everything(),
+    missing_text = "---"
```

Comment:
```suggestion
    missing_text = md("---")
```

and replace `context_missing_text` with `process_text`

Reply:
After testing, found that this cannot be done.